### PR TITLE
build: improve devcontainer.json

### PR DESCRIPTION
**Pull request by roninjin10**
Copy paste in some improvements to [the devcontainer from evmts](https://github.com/evmts/evmts-monorepo/blob/main/.devcontainer/devcontainer.json) added by @o-az

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -1,3 +1,36 @@
...
+        "esbenp.prettier-vscode"
+      ],
+      "settings": {
+        "solidity.monoRepoSupport": true,
+        "npm.packageManager": "pnpm",
+        // TODO get correct version
+        "solidity.compileUsingRemoteVersion": "v0.8.17",
+        "files.associations": {
+          ".gas-snapshot": "julia"
+        }
```
Comment (alvrs) [inline .devcontainer/devcontainer.json:32]: what's this one?

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -1,3 +1,36 @@
 {
-  "onCreateCommand": "curl -L https://foundry.paradigm.xyz | bash && /home/codespace/.foundry/bin/foundryup && pnpm install"
+  "onCreateCommand": "curl -L https://foundry.paradigm.xyz | bash && /home/codespace/.foundry/bin/foundryup && pnpm install",
+  "name": "latticexyz/mud Dev Container",
+  "features": {
+    "ghcr.io/devcontainers/features/git": {},
+    "ghcr.io/devcontainers/features/github-cli": {}
+  },
+  // TODO which other ports do we include here?
```
Comment (alvrs) [inline .devcontainer/devcontainer.json:8]: maybe the vite dev client port? (should be 3000)

File: .devcontainer/devcontainer.json [hunk 3]
```diff
@@ -1,3 +1,36 @@
...
+      "extensions": [
+        "JuanBlanco.solidity",
+        "dbaeumer.vscode-eslint",
+        "esbenp.prettier-vscode"
+      ],
+      "settings": {
+        "solidity.monoRepoSupport": true,
+        "npm.packageManager": "pnpm",
+        // TODO get correct version
+        "solidity.compileUsingRemoteVersion": "v0.8.17",
```
Comment (alvrs) [inline .devcontainer/devcontainer.json:29]: is it possible to have "logic" in this json that grabs the solidity version from somewhere?

**Review by alvrs — COMMENTED** [review]
very cool! generally down to merge this as is, just had some minor comments

**Comment by holic** [comment]
closing for now as this is out of date, feel free to open a new PR or reopen this one with updates

**Comment by roninjin10** [comment]
I have no memory of making this pr lol

**Comment by holic** [comment]
9 months in internet time is an eternity

**Pull request closed without merging**